api/src/test/java/org/pmiops/workbench/institution/InstitutionServiceTest.java

```diff
@@ -18,6 +18,7 @@
 import org.pmiops.workbench.db.model.DbVerifiedInstitutionalAffiliation;
 import org.pmiops.workbench.exceptions.ConflictException;
 import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.model.AgreementType;
```

Comment:
Please add these InstitutionServiceTests:
* validation success when agreementType is NULL and the contact email matches a domain
* something to demonstrate that agreementType is what determines how we validate email, rather than the presence of address or domains.  i.e. don't match against an email address if agreementType is MASTER

Reply:
there is already a test for that test_emailValidation_mismatch

Reply:
I will add more for Restricted and NULL

Reply:
Let's chat on Slack about point 2 here